### PR TITLE
Bug fix 3.6/invalid memory management in collect

### PR DESCRIPTION
**Pull request by mchacki**
Backport of https://github.com/arangodb/arangodb/pull/12271
Fixes: https://github.com/arangodb/arangodb/issues/12267

In case of Large values collected in multiple groups the memory management was incorrect, and a value could be freed although it is still in use. Which causes undefined behavior.

Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11196/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11205/

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,18 @@
 v3.6.6 (XXXX-XX-XX)
 -------------------
 
+* Fixed undefined behaviour in AQL COLLECT with multiple group variables.
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Fixed undefined behavior in AQL COLLECT with multiple group variables.
```

File: arangod/Aql/HashedCollectExecutor.cpp [hunk 2]
```diff
@@ -287,11 +287,35 @@ decltype(HashedCollectExecutor::_allGroups)::iterator HashedCollectExecutor::fin
...
-    _nextGroupValues.emplace_back(input.stealValue(reg.second));
+
+  if (_infos.getGroupRegisters().size() == 1) {
+    for (auto const& reg : _infos.getGroupRegisters()) {
+      // On a single register there can be no duplicate value
+      // inside the groupValues, so we cannot get into a situation
+      // where it is unclear who is responsible for the data
+      AqlValue a = input.stealValue(reg.second);
+      AqlValueGuard guard{a, true};
+      _nextGroupValues.emplace_back(input.stealValue(reg.second));
```
Comment (jsteemann) [inline arangod/Aql/HashedCollectExecutor.cpp:298]:
```suggestion
      _nextGroupValues.emplace_back(a);
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11213/

**Comment by KVS85** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr-linux/12899/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11220/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**